mnefun/_mnefun.py

```diff
@@ -2330,3 +2332,108 @@ def plot_reconstruction(evoked, origin=(0., 0., 0.04)):
     axs[2, 0].set_ylabel('Reconstruction')
     fig.tight_layout()
     return fig
+
+
+def plot_chpi_snr_raw(raw, win_length, n_harmonics):
+    """Compute and plot cHPI SNR from raw data
+
+    Parameters
+    ----------
+    win_length : int
+        Length of window to use for SNR estimates (seconds). A longer window
+        will naturally include more low frequency power, resulting in lower
+        SNR.
+    n_harmonics : int
+        Number of line frequency harmonics to include in the model.
+
+    Returns
+    -------
+    fig : Instance of matplotlib.figure.Figure
+        cHPI SNR as function of time, residual variance.
+    """
+
+    # plotting parameters
+    legend_fontsize = 10
+    legend_hspace = 25  # % of horizontal axis to reserve for legend
+
+    # get some info from fiff
+    sfreq = raw.info['sfreq']
+    linefreq = raw.info['line_freq']
+    linefreqs = (np.arange(n_harmonics + 1) + 1) * linefreq
+    buflen = int(win_length * sfreq)
+    if buflen <= 0:
+        raise ValueError('Window length should be >0')
+    (cfreqs, _, _, _, _) = _get_hpi_info(raw.info)
```

Comment:
I'll think about it. The fact that cHPI is off should also be obvious from the SNR plot. I'm not that keen on skipping buffers, since the user is probably interested in seeing the whole time range.